settings modal: Fix a few info density related issues

evykassirer opened the pull request:
Changes:

* Scale width of "general/personal/subscriber" tabs with font size (and similar for user groups)
* Scale X button on the discard button with font size
* Left pane is always 40% width and right pane is always 60% width (previously it was 50/50 at wider screen sizes). This simplifies some CSS and also ensures there's enough space at larger font-sizes for the save/discard buttons to appear beside setting section headings without overflowing onto the next line (which was happening when the X icon got bigger)

**screenshots of subscriptions at 12/14/16/20 px**

| before | after |
| --- | --- |
| ![Screen Shot 2025-02-20 at 15 13 53](https://github.com/user-attachments/assets/7cd8d9b3-a344-466e-8182-a7aba113bc86) | ![Screen Shot 2025-02-20 at 15 21 29](https://github.com/user-attachments/assets/65f6224d-8221-44f3-9caa-2f97559c1496) |
| ![Screen Shot 2025-02-20 at 15 13 40](https://github.com/user-attachments/assets/7c4c6fd4-1651-438d-8cdf-74f497080e6d) | ![Screen Shot 2025-02-20 at 15 21 15](https://github.com/user-attachments/assets/a7141c19-d5c9-4ceb-994d-5ddc56db41a3) |
| ![Screen Shot 2025-02-20 at 15 13 20](https://github.com/user-attachments/assets/10c2639f-fad5-4192-ae04-144299d5c6e9) | ![Screen Shot 2025-02-20 at 15 21 00](https://github.com/user-attachments/assets/10d89b1d-35cc-4dec-a741-c08881c2bede) |
| ![Screen Shot 2025-02-20 at 15 13 07](https://github.com/user-attachments/assets/c00978d6-b72c-4ece-8599-ea76a0bdf04a) | ![Screen Shot 2025-02-20 at 15 20 45](https://github.com/user-attachments/assets/f1697e71-a2a6-4283-8982-b90c0edf07f7) |

**screenshots of usergroups at 12/14/16/20 px**

| before | after |
| --- | --- |
| ![Screen Shot 2025-02-20 at 15 26 09](https://github.com/user-attachments/assets/679fa4e3-35c7-4787-9d7e-e5f706b9d757) | ![Screen Shot 2025-02-20 at 15 23 27](https://github.com/user-attachments/assets/94aeef96-9d7d-4403-bdba-80cd1feabe2f) |
| ![Screen Shot 2025-02-20 at 15 25 51](https://github.com/user-attachments/assets/c0a72a5e-1463-423d-8c3a-6813914503d3) | ![Screen Shot 2025-02-20 at 15 24 04](https://github.com/user-attachments/assets/f8c032fc-e9f2-4d2c-8c00-f4aff5866412) |
| ![Screen Shot 2025-02-20 at 15 25 35](https://github.com/user-attachments/assets/196cde7a-d419-4676-b38b-57838990eefe) | ![Screen Shot 2025-02-20 at 15 24 14](https://github.com/user-attachments/assets/3cc6870a-afbc-44dd-8f91-106d2fb39a41) |
| ![Screen Shot 2025-02-20 at 15 25 14](https://github.com/user-attachments/assets/d448a72d-0d62-4516-800b-9aec039ed3eb) | ![Screen Shot 2025-02-20 at 15 24 31](https://github.com/user-attachments/assets/6c0714ff-a5db-4d67-a770-dadf9e778de8) |

alya commented:
Feels good to me! @sahil839 any feedback on this one?

I flagged [#issues > UI issues in channel settings (narrow windows)](https://chat.zulip.org/#narrow/channel/9-issues/topic/UI.20issues.20in.20channel.20settings.20.28narrow.20windows.29/with/2105438) while reviewing, but it's not caused by this PR.

sahil839 reviewed (commented):
- Text and icon for discard button seems misaligned for some font sizes like 14px and 18px.
![Screenshot from 2025-02-27 14-33-09](https://github.com/user-attachments/assets/12634a44-8647-4c60-8a70-f761dfb80311)
- On narrow screen at 16px. Is there some other way to handle this when screen space is limited? Maybe breakpoints for hiding left panel can be changed.
![Screenshot from 2025-02-27 14-34-05](https://github.com/user-attachments/assets/633f5bb1-d404-4f49-b292-3900864031c5)

evykassirer commented on zulip-py3-venv:
```diff
@@ -0,0 +1 @@
+/srv/zulip-venv-cache/270ffd9b5616da7e0b0dfcea43f03d011c61be54/zulip-py3-venv
```
Oh oops, thanks! 🙈

evykassirer commented:
> Text and icon for discard button seems misaligned for some font sizes like 14px and 18px.

I'm having trouble reproducing this (see screenshots in the PR description for what it looks like for me). What browser are you using? Also out of curiosity what line height?

evykassirer commented:
Pushing a new commit to wrap the buttons to be on a second line on narrow screens (cc @alya who reported this elsewhere). It's not the most beautiful solution but it feels fine given that this area of the UI isn't redesigned yet.

![Kapture 2025-03-03 at 12 57 08](https://github.com/user-attachments/assets/1f21ac7d-26c3-4979-a1bf-fdacd7e37227)

sahil839 commented:
> I'm having trouble reproducing this (see screenshots in the PR description for what it looks like for me). What browser are you using? Also out of curiosity what line height?

Seems fine for 14px, but still looks slighty misaligned for 18px at 140 line height in Chrome.

sahil839 commented:
It is mentioned in [#issues > 🎯 UI issues in channel settings (narrow windows) @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/.F0.9F.8E.AF.20UI.20issues.20in.20channel.20settings.20.28narrow.20windows.29/near/2106698) that the left panel issue is fixed, but this still can be reproduced for screen width around 500px and less than that.
![Screenshot from 2025-03-04 08-42-10](https://github.com/user-attachments/assets/71663207-5120-4e7c-b64a-9d9e91fc50ba)

evykassirer commented:
This is what I see at 18px/140 in Chrome:

<img width="684" alt="image" src="https://github.com/user-attachments/assets/e7107e6e-d12d-42b1-a33b-1101906a4cdd" />


> It is mentioned in [#issues > 🎯 UI issues in channel settings (narrow windows) @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/.F0.9F.8E.AF.20UI.20issues.20in.20channel.20settings.20.28narrow.20windows.29/near/2106698) that the left panel issue is fixed, but this still can be reproduced for screen width around 500px and less than that. ![Screenshot from 2025-03-04 08-42-10](https://private-user-images.githubusercontent.com/35494118/418818888-71663207-5120-4e7c-b64a-9d9e91fc50ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDExNDEzMjgsIm5iZiI6MTc0MTE0MTAyOCwicGF0aCI6Ii8zNTQ5NDExOC80MTg4MTg4ODgtNzE2NjMyMDctNTEyMC00ZTdjLWI2NGEtOWQ5ZTkxZmM1MGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA1VDAyMTcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwMzM0ZWM5OWI1YTc1OTU1MzBkNTAxM2Q0Y2U0ZDIyN2NhNmU0MWEzN2QzZTUyOGUwMGM2MGFiYmE2OGI4ZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gDiwO2Kz64tIPHEcn-RWp9uy2kUx1kL9nEyDbTxWJWA)

Ah I had forgotten about this narrow window case, thanks! Here's a fix

```diff
diff --git a/web/styles/subscriptions.css b/web/styles/subscriptions.css
index 8ba7c1dd2d..a79b3f8767 100644
--- a/web/styles/subscriptions.css
+++ b/web/styles/subscriptions.css
@@ -343,8 +343,12 @@ h4.user_group_setting_subsection_title {
    breakpoint that collapses the left pane, but still narrow
    enough that the sort buttons need a second row to be visible
    without overlapping other UI. Note that for 16px and larger
-   font sizes, we always show two rows of buttons. */
-@container settings-overlay (width >= $settings_overlay_sidebar_collapse_breakpoint) {
+   font sizes, we always show two rows of buttons.
+
+   We also show two rows of buttons on particularly narrow
+   screens where the full overlay doesn't have enough space
+   for all the buttons. */
+@container settings-overlay (width >= $settings_overlay_sidebar_collapse_breakpoint) or (width <= 36em) {
     @container subscriptions (width < calc(80em + 80px)) {
         #subscription_overlay .subscriptions-container {
             .left {

```

sahil839 commented:
I can reproduce the save-discard alignment bug only when window width is greater than 830px, might not be the exact breakpoint, but the alignment seems fine for widths less than 830px.

karlstolley commented on web/styles/subscriptions.css:
```diff
@@ -339,11 +340,16 @@ h4.user_group_setting_subsection_title {
+   screens where the full overlay doesn't have enough space
+   for all the buttons. */
+@container settings-overlay (width >= $settings_overlay_sidebar_collapse_breakpoint) or (width <= 36em) {
+    @container subscriptions (width < calc(80em + 80px)) {
```
I'm still uneasy about this compound query-within-a-query. It seems like we should be able to definite a container independent of `settings-overlay` whose width can be queried as a simple less-than condition.

evykassirer replied:
I was able to make things simpler because the two container queries were on elements with similar width, but:

(1) `settings-overlay` needs to be the `@container`, not `subscriptions-container`. The `subscriptions-container` is part of the CSS declaration inside the `@container` block and it doesn't seem to work unless the `subscriptions-container` is inside the `@container` classname. Maybe I'm missing something, let me know if so!

(2) There needs to be two conditions, one for the wider two-panel view and one for the narrower one-panel view. Both depend on the size of the left panel, but the `@container` query needs to be on the parent which contains the right panel which also needs to change. One possible alternative is two different queries for the left panel and right panel, but that seems overall more confusing to me.

evykassirer commented:
> I can reproduce the save-discard alignment bug only when window width is greater than 830px, might not be the exact breakpoint, but the alignment seems fine for widths less than 830px.

I tried at several widths but couldn't reproduce. Could you see if you can find an alignment fix on your end and I can incorporate that?

sahil839 commented:
Ok. I will try to fix that.

Also, noticed this -
![Screenshot from 2025-03-07 12-52-17](https://github.com/user-attachments/assets/b97c2a6f-d70e-4e59-baee-cfefa4901f20)

sahil839 commented:
I did the following change -
```diff
diff --git a/web/styles/app_components.css b/web/styles/app_components.css
index f50c63721e..9f7247d843 100644
--- a/web/styles/app_components.css
+++ b/web/styles/app_components.css
@@ -955,6 +955,10 @@ input.settings_text_input {
             margin-right: 3px;
             font-size: 1.0714em; /* 15px at 14px/em */
             font-weight: 400;
+
+            &:before {
+                vertical-align: middle;
+            }
         }
 
         .save-discard-widget-button-text {
```

This fixes the case I shared above, but results in icon being slightly above at some screen width. Maybe you can check once.
![image](https://github.com/user-attachments/assets/831916ac-2f21-4006-a034-df5df4afd1d0)

evykassirer commented:
Updates:
* Rebased and resolve conflicts
* Removed the left margin from headings
* Instead of doing your change, I tried adding more flex and center alignment to the icons. Does that have a similar effect for you?

The slightly above alignment is hard to fix, and I think it's due to the icon's height itself (similar to our issues with emoji alignment). I think this is fine given that we'll want to redesign this UI and use a proper icon in the future.

sahil839 commented:
> Instead of doing your change, I tried adding more flex and center alignment to the icons. Does that have a similar effect for you?

I can still reproduce this without doing any changes in the PR.
![image](https://github.com/user-attachments/assets/4a3c3aaa-0263-4e50-b8ba-a59bc7cb69de)

sahil839 commented:
Yes, issue seems due to height. Will setting the icon height help?

evykassirer commented:
> Yes, issue seems due to height. Will setting the icon height help?

That doesn't seem to make any difference.

evykassirer commented:
> I can still reproduce this without doing any changes in the PR.

Does adding the same `vertical-align: middle;` on top of the current state of the PR still fix it for you?

sahil839 commented:
> > I can still reproduce this without doing any changes in the PR.
> 
> Does adding the same `vertical-align: middle;` on top of the current state of the PR still fix it for you?

No.

evykassirer commented:
@timabbott how would you feel about integrating this PR as is, and doing this icon twiddling as a followup issue? I'd like to make sure the rest of it gets enough burn in on CZO, and I'm having trouble getting the last bit fixed in a way that works for me locally as well as for Sahil (whose issue I can't reproduce locally).

timabbott commented:
Yeah, I think it's probably better to merge this and fix that forward. Please make sure that save/discard icon issue is tracked as a release goal, maybe just start a new issues thread with a target on it, since I think weird quirks like this benefit from visibility.
